enarx-keep-sgx-shim/src/start.S

```diff
@@ -76,9 +102,10 @@ enclave:
     cmp     $0,                     %rax            # If CSSA > 0...
     jne     .Levent                                 # ... restore stack from AEX[CSSA-1].
 
-    # Clear unused registers, set the stack pointer and jump to Rust
+    # Clear unused registers and xCPU state, set the stack pointer and jump to Rust
     zerop
     zerot
+    zerox
```

Comment:
This doesn't work. `zerox` pushes `%rax` and `%rdx` to the stack. But you don't setup the secure stack until below. Therefore, you can't use the stack yet.

Reply:
@npmccallum Would it make sense to move this two lines down after `mov  STACK(%rcx),  %rsp`?

Reply:
I suspect it might be more readable to move the stack setup line up to the top and do all the zeroing afterwards.